### ISSUE TITLE
Events With Multiple Local Handlers Can Be Interfered By Each Other

### ISSUE THREAD

**Issue by melvinkcx** [melvinkcx/fastapi-events#16]
Today, an event with multiple local handlers registered can interfere with each other as all handlers are given the same shared copy of events. 

For instance,

```
# anywhere in code

dispatch(Events.USER_CREATED, {"key_in_payload": 123})
```

```
# in handlers.py

@local_handler.register(event_name=Events.USER_CREATED)
async def handle_user_created(event: Event):
    _, payload = event
    payload.pop("key_in_payload")


@local_handler.register(event_name=Events.USER_CREATED)
async def handle_user_created_2(event: Event)
    _, payload = event
    payload["key_in_payload"]  # KeyError
```

## Proposal

A copy of the payload should be passed to the handlers.

**Comment by danielhasan1** [comment]
The proposal will solve this issue on a small level, but when the payload size is big and there are multiple consumers who are consuming the dispatched event at the same time this will become a big issue it will block all the memory. Especially for long-running tasks or coroutines.
A quick fix that comes into my mind is writing a wrapper inside dispatch to make the payload a frozen dict.
multiple handlers could accept it, frozen dict won't change. if anyone wants to change any property of that dict they could create a separate copy of that dict or get items from that dict.

**Comment by melvinkcx** [comment]
@danielhasan1 you're right. I haven't tried to address this issue because the `payload` can technically be of any type, not just `dict`. Either approach risks breaking the user's code if the payload contains values (custom objects, etc) that cannot be cloned or frozen. 

I'm also not aware of any "standard" approach to making a `dict` immutable as [PEP 416](https://www.python.org/dev/peps/pep-0416/) was rejected and [PEP 603](https://peps.python.org/pep-0603/) is still under drafting stage. 

I think it's best we keep this unfixed for now.

**Comment by ebrahimradi** [comment]
I think it's not about this library, it's about our design. events should be immutable. as a developer you should try design to have immutable events. for example you can try pydantic frozen classes or frozen dataclass or something that guarantee immutable events.
one suggestion is to change library to only accept immutable events data :D
